### PR TITLE
Use SPDX license expression in project metadata

### PR DESCRIPTION
**Pull request by RazerM**
As a downstream user it is desirable to be able to programmatically determine the precise licenses used by our dependencies. The emerging convention for this is the [SPDX License List](https://spdx.org/licenses/). SPDX License Expressions are already used in the JavaScript (npm) and Rust (crates.io) ecosystems, for example.

In Python, there is [PEP 639](https://peps.python.org/pep-0639/) which would add a standard 'License-Expression' metadata field. The discussion around this PEP seems to have gone stale in 2021.

This merge request is in lieu of that PEP coming soon. Unless you as project maintainers think the proposed change in this PR has downsides (which you are entitled to!), I would ask that you accept this change so that it's possible for users to attempt to parse your license metadata as an SPDX License Expression.

### CODE REVIEW

**Comment by nejch** [comment]
Thanks for this @RazerM, I know the pain with unparseable licenses for SBOMs/OSS notices and SPDX makes this more sane, just missed this :)

We were actually going to switch to PEP 621 already (draft here https://github.com/python-gitlab/python-gitlab/blob/8d45db78b4e848b0ed41cc1ce338457b0ebc81bc/pyproject.toml#L39) and that leaves this intentionally a bit open by allowing either file or text keys, and we went for file there. Let me just check what the best approach for pep621 is. Otherwise if we go this way this'll just need a little `git commit --amend` as we use conventional commits (see CONTRIBUTING.rst).

**Comment by RazerM** [comment]
Thanks for taking this into consideration

I totally get the PEP 621 switch — I've been doing it myself for several projects and it's a joy to finally be using this nice file format :-)

The part of that PEP which worries me a bit around this license stuff is

> A practical string value for the license key has been purposefully left out to allow for a > future PEP to specify support for [SPDX](https://spdx.dev/) expressions

Because it means you might have projects which provide a license file only (because the PEP disallows `file` *and* `text`) and there's no way for downstream users to find out which license it is programmatically without trying to parse the file. (Until PEP 639 arrives, at least).

In practice though, things are not so bad for setuptools, which you are still using in the linked pyproject.toml. I can confirm that if you use `license = { text = "LGPL-3.0-only" }`, setuptools still installs your `COPYING` file which I think might be the best of both worlds.

If you try it out you should see e.g. `lib/python3.10/site-packages/python_gitlab-3.11.0.dist-info/COPYING` exists.

An alternative I could imagine that would still let you use `license = { file = "COPYING" }` would be to add `SPDX-License-Identifier: LGPL-3.0-only` somewhere in the `COPYING` file itself?

**Comment by nejch** [comment]
That's ok, it's still closer to the current `setup.py` text in any case :) I guess some of the reluctance in PEP 639 comes from this uncertainty - here as well, the current maintainers are not the authors so to me, we're basically disambiguating it right now in this PR and interpreting that the original author explicitly wanted `-only` and not `-or-later` when choosing the classifier, it's not entirely clear to me from reading https://github.com/python-gitlab/python-gitlab/issues/447 :)

Could you just amend your commit message according to https://www.conventionalcommits.org/en/v1.0.0/#summary? we use it for our cahngelogs. This will need a force-push after as well.

**Comment by RazerM** [comment]
Will do, would you consider this a `chore`?

**Comment by nejch** [comment]
I think chore is fine here yes unless you'd prefer to have this explicitly in the changelog as a fix.

File: setup.py
Comment (nejch) [inline setup.py:28]: @max-wittig WDYT before I merge this, we don't have a `This project is licensed under the terms of the <something> license.` so from what I can tell it's not even clear (even from Gauvain's initial comments and text), and we should decide right here on `LGPL-3.0-only` vs `LGPL-3.0-or-later`.

The proposed PEP even doesn't want to disambiguate this automatically (https://peps.python.org/pep-0639/#mapping-license-classifiers-to-spdx-identifiers).

Comment (max-wittig) [reply setup.py:28]: I guess or later is fine?

Comment (nejch) [reply setup.py:28]: Works for me as well.

```suggestion
    license="LGPL-3.0-or-later",
```

Comment (nejch) [reply setup.py:28]: @RazerM are you still interested and would like to apply the suggested change above?

Comment (RazerM) [reply setup.py]: Done

**Review by nejch — APPROVED** [review]

File: setup.py
Comment (nejch) [reply setup.py]: Thanks for the PR and the discussion above!

**Pull request merged**
